docs-new/docs/graphql/core/how-it-works/metadata-schema.mdx

```diff
@@ -0,0 +1,192 @@
+database _(which is by default the same database from which data is
+served over the GraphQL API if it is Postgres)_. When initialized, the
+Hasura GraphQL engine creates a schema called `hdb_catalog` in the
+metadata database and initializes a few tables under it as described
+below.
+
+## hdb_catalog schema
+
+This schema is created by the Hasura GraphQL engine to manage its
+internal state. Whenever a table/permission/relationship is
+created/updated using the Hasura console or the metadata API, the Hasura
+GraphQL engine captures that information and stores it in the
+corresponding tables.
+
+The following tables are used by the Hasura GraphQL engine:
+
+### hdb_table table
+
+This table stores information about all the tables/views which are
+created/tracked using the Hasura console or the metadata API.
+
+#### Schema
+
+<Thumbnail
+  src="/img/graphql/core/engine-internals/hdb_table.jpg"
+  alt="Schema for hdb_table table"
+  width="30%"
+/>
+
+#### Column Definitions
+
+| column            | description                                                                                                                                           |
+|-------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------|
+| table_schema      | Captures information about the schema under which a table/view is tracked                                                                             |
+| table_name        | Captures name of the tracked table/view.                                                                                                              |
+| is_system_defined | If it is true, then the table/view is created by GraphQL engine for internal purpose. If it is false, then the table/view is created by the end user. |
+
+### hdb_relationship table
+
+This table stores information about the relationships created for
+tables/views using the Hasura console or the metadata API.
+
+#### Schema
+
+<Thumbnail
+  src="/img/graphql/core/engine-internals/hdb_relationship.jpg"
+  alt="Schema for hdb_relationship table"
+  width="30%"
+/>
+
+#### Column Definitions
+
+<table>
+  <thead>
+    <tr>
+        <th width="19%">column</th>
+        <th width="80%">description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>table_schema</td>
+      <td>Captures information about the schema under which a relationship is created.</td>
+    </tr>
+    <tr>
+      <td>table_name</td>
+      <td>Captures name of the table/view under which a relationship is created.</td>
+    </tr>
+    <tr>
+      <td>rel_name</td>
+      <td>Captures name of the relationship.</td>
+    </tr>
+    <tr>
+      <td>rel_type</td>
+      <td>Captures the permission type (insert/select/update/delete).</td>
+    </tr>
+    <tr>
+      <td>perm_def</td>
+      <td>
+      Captures information about how the relationship is defined. For example:
```

Comment:
Missing code block. Please cross check with original `rst` file to ensure no content is missed.

Reply:
Ack!